Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -4,162 +4,142 @@
 {
     /// <summary>
-    /// Based on sample custom skill provided in Azure Search. Provided a user-defined list of entities
-    /// this function determines the Entity first occurrence within a given document. This list of entities
-    /// must repeatedly be provided by the user for each document.
+    /// Based on sample custom skill provided in Azure Search. Given a user-defined list of entities,
+    /// this function will find all occurences of that entity in some input text.
     /// </summary>
     public static class CustomEntitySearch
     {
-        // Use this to load from "csv" or "json" file 
-        public static IList<string> preLoadedWords = null;
+        // ** Some global variables used for configuration.**
+        // ** Change these values prior to deploying your function **
+        public static bool ExactMatchesShouldBeCaseSensitive = false;
+        public static string EntityDefinitionLocation = "words.csv"; // other option is "words.json"
+        public static readonly int MaxRegexEvalTimeInSeconds = 10;
 
-        private static readonly int MaxRegexEvalTime = 1;
-        private static bool substringMatch = false;
+
+        /// Entity definition is lazy loaded on first function call. This may result in the
+        /// first function call taking several seconds (since it needs to parse entity definitions).
+        /// subsequent function calls should be significantly faster
+        private static WordLinker _userDefinedEntities = null;
+        private static Regex _precompiledExactMatchRegex = null;
 
         /// <summary>
-        /// We assert the following assumptions:
-        /// 1. All text files contain characters with unicode encoding
-        /// 2. Words can contain special characters and numbers
-        /// 3. The provided entities are not case sensitive
+        /// Find instances of custom entities from either words.csv or words.json
+        /// in input text
         /// </summary>
         [FunctionName("custom-entity-search")]
         public static async Task<IActionResult> RunCustomEntitySearch(
             [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
             ILogger log,
             ExecutionContext executionContext)
         {
-            if (preLoadedWords == null)
+            log.LogInformation("Custom Entity Search function: C# HTTP trigger function processed a request.");
+
+            if (_userDefinedEntities == null || _precompiledExactMatchRegex == null
+                || executionContext.FunctionName == "unitTestFunction") // always reload data for tests
             {
-                preLoadedWords = WordLinker.WordLink(executionContext.FunctionAppDirectory, "csv").Words;
+                _userDefinedEntities = WordLinker.WordLink(EntityDefinitionLocation);
+                _precompiledExactMatchRegex = new Regex($@"\b({string.Join("|", _userDefinedEntities.Words)})\b",
```

Comment:
It looks like it's possible words would contain regex special characters. They should probably be escaped first.